eth/tracers/internal/tracetest/flat_calltrace_test.go

```diff
@@ -0,0 +1,220 @@
+type flatCallTracerTest struct {
+	Genesis      core.Genesis    `json:"genesis"`
+	Context      callContext     `json:"context"`
+	Input        string          `json:"input"`
+	TracerConfig json.RawMessage `json:"tracerConfig"`
+	Result       []flatCallTrace `json:"result"`
+}
+
+func flatCallTracerTestRunner(tracerName string, filename string, dirPath string, t testing.TB) error {
+	// Call tracer test found, read if from disk
+	blob, err := os.ReadFile(filepath.Join("testdata", dirPath, filename))
+	if err != nil {
+		return fmt.Errorf("failed to read testcase: %v", err)
+	}
+	test := new(flatCallTracerTest)
+	if err := json.Unmarshal(blob, test); err != nil {
+		return fmt.Errorf("failed to parse testcase: %v", err)
+	}
+	// Configure a blockchain with the given prestate
+	tx := new(types.Transaction)
+	if err := rlp.DecodeBytes(common.FromHex(test.Input), tx); err != nil {
+		return fmt.Errorf("failed to parse testcase input: %v", err)
+	}
+	signer := types.MakeSigner(test.Genesis.Config, new(big.Int).SetUint64(uint64(test.Context.Number)))
+	origin, _ := signer.Sender(tx)
+	txContext := vm.TxContext{
+		Origin:   origin,
+		GasPrice: tx.GasPrice(),
+	}
+	context := vm.BlockContext{
+		CanTransfer: core.CanTransfer,
+		Transfer:    core.Transfer,
+		Coinbase:    test.Context.Miner,
+		BlockNumber: new(big.Int).SetUint64(uint64(test.Context.Number)),
+		Time:        uint64(test.Context.Time),
+		Difficulty:  (*big.Int)(test.Context.Difficulty),
+		GasLimit:    uint64(test.Context.GasLimit),
+	}
+	_, statedb := tests.MakePreState(rawdb.NewMemoryDatabase(), test.Genesis.Alloc, false)
+
+	// txctx := &tracers.Context{
+	// 	BlockHash: blockHash,
+	// 	TxIndex:   task.index,
+	// 	TxHash:    txs[task.index].Hash(),
+	// }
+
+	// Create the tracer, the EVM environment and run it
+	tracer, err := tracers.DefaultDirectory.New(tracerName, new(tracers.Context), test.TracerConfig)
+	if err != nil {
+		return fmt.Errorf("failed to create call tracer: %v", err)
+	}
+	evm := vm.NewEVM(context, txContext, statedb, test.Genesis.Config, vm.Config{Debug: true, Tracer: tracer})
+
+	msg, err := tx.AsMessage(signer, nil)
+	if err != nil {
+		return fmt.Errorf("failed to prepare transaction for tracing: %v", err)
+	}
+	st := core.NewStateTransition(evm, msg, new(core.GasPool).AddGas(tx.Gas()))
+
+	if _, err = st.TransitionDb(); err != nil {
+		return fmt.Errorf("failed to execute transaction: %v", err)
+	}
+
+	// Retrieve the trace result and compare against the etalon
+	res, err := tracer.GetResult()
+	if err != nil {
+		return fmt.Errorf("failed to retrieve trace result: %v", err)
+	}
+	ret := new([]flatCallTrace)
+	if err := json.Unmarshal(res, ret); err != nil {
+		return fmt.Errorf("failed to unmarshal trace result: %v", err)
+	}
+	if !jsonEqualFlat(ret, test.Result) {
+		t.Logf("tracer name: %s", tracerName)
+
+		// uncomment this for easier debugging
+		have, _ := json.MarshalIndent(ret, "", " ")
```

Comment:
Ah I think I uncommented these for debugging and forgot to comment them again. Can you please comment before merging?

Reply:
Resolved by https://github.com/ethereum/go-ethereum/pull/26377/commits/bb887f3e1c6b8e95758db14ef9b47572c4da6458